gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/TXState.java

```diff
@@ -1347,6 +1391,101 @@ final void cleanupTXRS(final TXRegionState[] regionStates,
     }
   }
 
+  //rollback can be done locally for snapshot just as commit is done
+  private void rollBackUncommittedEntries() throws RegionClearedException {
+    // we need to take entry lock on primary otherwise the following can happen
+    // on primary : we replace uncommitted entry with committed entry
+    // on primary new write comes, which goes to secondary
+    // on secondary : we replace the new entry with old committed entry
+    // to avoid this we can compare the version or keep the uncommitted RE reference too in
+    // the txStateand don't conflict on secondary.
+
+    Iterator<RegionEntry> itr1 = unCommittedEntryReference.iterator();
+    Iterator<Object> itr2 = committedEntryReference.iterator();
+    Iterator<LocalRegion> itr3 = regionReference.iterator();
+
+    GemFireCacheImpl.getInstance().getLogger().info("Rolling back the changes.. " + unCommittedEntryReference.size()
```

Comment:
You want to keep this in info. Specially since conflict is added, this might fill the logs. No?